src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -190,10 +218,16 @@ public static string GetEnvironmentFileName(string fileName, string environmentV
         return $"{fileName}.{environmentValue}{CONFIG_EXTENSION}";
     }
 
-    public bool DoesFileExistInCurrentDirectory(string fileName)
+    /// <summary>
+    /// Checks if the file exists in the directory.
+    /// Works for both relative and absolute paths.
+    /// </summary>
+    /// <param name="filePath"></param>
+    /// <returns>True if file is found, else false.</returns>
+    public bool DoesFileExistInDirectory(string filePath)
     {
         string currentDir = _fileSystem.Directory.GetCurrentDirectory();
-        return _fileSystem.File.Exists(_fileSystem.Path.Combine(currentDir, fileName));
+        return _fileSystem.File.Exists(_fileSystem.Path.Combine(currentDir, filePath));
```

Comment:
Originally, the plan was to stop using `Path.Combine`. However, after further inspection of [documentation]https://learn.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-7.0#system-io-path-combine(system-string-system-string)()
> This method is intended to concatenate individual strings into a single string that represents a file path. However, if an argument other than the first contains a rooted path, any previous path components are ignored, and the returned string begins with that rooted path component.


`Path.Combine(currentDir, filePath)` resolves to `"C:\dab\configs\dab-config.json"` for
- currentDir == `"C:\dab\configs\"`
- filePath == `"C:\dab\configs\dab-config.json"` or `"configs\dab-config.json"` or `"dab-config.json"`

if `filePath` is a rooted path different that the first param of `currentDir`, the rooted path in `filePath` takes precedence (Per docs):

So, `Path.Combine(currentDir, filePath)` resolves to `"D:\dab2\configfiles\dab-config.json"` for
- currentDir == `"C:\dab\configs\"`
- filePath == `"D:\dab2\configfiles\dab-config.json"`

Reply:
Thanks for the details and the link.